### ISSUE TITLE
Self-Modifying Code

### ISSUE THREAD

**Issue by johannesmono** [Vector35/debugger#116]
Is there any way to make self-modifying code work with the debugger plugin? Thank you!

**Comment by the-wondersmith** [comment]
I can't second this hard enough. Alternatively there were a way to perhaps *guide* BN through making the same modifications the code itself would? 

I'm currently fiddling with a Windows binary that was packed with PECompact v3. At about four instructions in from the entry point, eax is xor'd to set its value to `00000000` which is then passed as eip which of course throws an exception. Said exception is caught via SEH, wherein a bunch of stuff appears to happen ending with eax being set to the correct value and execution then proceeds.

As far as BN can tell though, the binary is just.... FUBAR'd.

**Comment by psifertex** [comment]
Certainly. You can definitely use BN to write new bytes directly into its binaryview based on the unpacked file contents. Just use the debugger view to copy bytes as hex and paste them into the binary view. There's no automation that does it automatically, but it should work fine otherwise.
